File: AspNetMvcUnleashed/0672329980_source/Chapter03/CS/MvcApplication1/MvcApplication1/Controllers/CatalogController.cs
using System.Web.Mvc;

namespace MvcApplication1.Controllers
{
    public class CatalogController : Controller
    {
       
        public ActionResult Create()
        {
            return View();
        }

        public ActionResult Delete(int id)
        {
            return View();
        }


        protected override void HandleUnknownAction(string actionName)
        {
            ViewData["actionName"] = actionName;
            View("Unknown").ExecuteResult(this.ControllerContext);
        }


    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter03/VB/MvcApplication1/MvcApplication1/Controllers/CatalogController.vb
Public Class CatalogController
    Inherits System.Web.Mvc.Controller


    Function Create() As ActionResult
        Return View()
    End Function

    Function Delete() As ActionResult
        Return View()
    End Function

    Protected Overrides Sub HandleUnknownAction(ByVal actionName As String)
        ViewData("actionName") = actionName
        View("Unknown").ExecuteResult(Me.ControllerContext)
    End Sub


End Class
